frontend/app/applications/[id]/ApplicationHeader.js

```diff
@@ -57,25 +118,20 @@ export default function ApplicationHeader({ application }) {
             fontSize: { xs: '1rem', sm: '1.5rem' },
           }}
         >
-          {application.salary !== undefined && application.salary !== null ? (
-            application.salary === 0 ? (
+          {Number(application.salary || 0) === 0 ? (
```

Comment:
Following is better, read about `||` vs `??` operators.
```suggestion
          {Number(application.salary ?? 0) === 0 ? (
```